test/lib/requests.go

```diff
@@ -0,0 +1,269 @@
+// it can successfully talk to the REST API.  Times out after 4 minutes
+func newEventGetter(podName string, client *Client, logf logging.FormatLogger) (eventGetterInterface, error) {
+	egi := &eventGetter{podName: podName, podNamespace: client.Namespace, kubeClientset: client.Kube.Kube, podPort: RecordEventsPort, logf: logf}
+	err := egi.forwardPort()
+	if err != nil {
+		return nil, err
+	}
+
+	err = egi.waitTillUp()
+	if err != nil {
+		return nil, err
+	}
+	return egi, nil
+}
+
+// Get information about the provided podName.  Uses list (rather than get) and
+// returns a pod list for compatibility with the monitoring.PortForward
+// interface
+func (eg *eventGetter) getRunningPodInfo(podName, namespace string) (*v1.PodList, error) {
+	pods, err := eg.kubeClientset.CoreV1().Pods(namespace).List(metav1.ListOptions{FieldSelector: fmt.Sprintf("metadata.name=%s", podName)})
+	if err == nil && len(pods.Items) != 1 {
+		err = fmt.Errorf("no %s Pod found on the cluster", podName)
+	} else if pods.Items[0].Status.Phase != corev1.PodRunning {
+		err = fmt.Errorf("pod %s in state %s, wanted Running", podName, pods.Items[0].Status.Phase)
+	}
+
+	return pods, err
+}
+
+// Try to forward the pod port to a local port somewhere in the range 30000-60000.
+// keeps retrying with random ports in that range for 4 minutes.
+func (eg *eventGetter) forwardPort() error {
+	portRand := rand.New(rand.NewSource(time.Now().UnixNano()))
+	portMin := 30000
+	portMax := 60000
+	var internalErr error
+
+	wait.PollImmediate(minEvRetryInterval, timeoutEvRetry, func() (bool, error) {
+		localPort := portMin + portRand.Intn(portMax-portMin)
+		if err := monitoring.CheckPortAvailability(localPort); err != nil {
+			internalErr = err
+			return false, nil
+		}
+		pods, err := eg.getRunningPodInfo(eg.podName, eg.podNamespace)
+		if err != nil {
+			internalErr = err
+			return false, nil
+		}
+
+		pid, err := monitoring.PortForward(eg.logf, pods, localPort, eg.podPort, eg.podNamespace)
+		if err != nil {
+			internalErr = err
+			return false, nil
+		}
+		internalErr = nil
+
+		eg.forwardPID = pid
+		eg.port = localPort
+		eg.host = "localhost"
+		return true, nil
+	})
+	if internalErr != nil {
+		return fmt.Errorf("timeout forwarding port: %s", internalErr)
+	}
+	return nil
+}
+
+// Return the min available, max seen by the recordevents pod.
+// maxRet is the largest event that has ever been seen (whether it's been trimmed
+// or not).  minRet is the smallest event still available via Get, or 1+maxRet if
+// no events are available.  maxRet starts at 0 when no events have been seen.
+func (eg *eventGetter) getMinMax() (minRet int, maxRet int, errRet error) {
+	resp, err := http.Get(fmt.Sprintf("http://%s:%d%s", eg.host, eg.port, GetMinMaxPath))
+	if err != nil {
+		return -1, -1, fmt.Errorf("http get error: %s", err)
+	}
+	defer resp.Body.Close()
+	bodyContents, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return -1, -1, fmt.Errorf("Error reading response body %w", err)
```

Comment:
Error strings should not be capitalized, see https://github.com/golang/go/wiki/CodeReviewComments#error-strings.
The same for other error strings.

Reply:
changed throughout